lib/castle.ex

```diff
@@ -1,2 +1,24 @@
 defmodule Castle do
+
+  defmodule Interval do
```

Comment:
I moved the Interval/Bucket/Grouping structs out of the BigQuery namespace, and into Castle.  As they now have nothing to do with BigQuery.

(I left the group struct/tests in here, even though the API endpoints are gone for now).